### ISSUE TITLE
Low vulnerability | meow ->  yargs-parser ^10.0.0

### ISSUE THREAD

**Issue by rackstar** [hyper-expanse/semantic-delivery-gitlab#40]
Dependency vulnerability: yargs-parser
Patched in: >=13.1.2 <14.0.0 || >=15.0.1 <16.0.0 || >=18.1.2

@hutson/semantic-delivery-gitlab > conventional-commits-detector > meow > yargs-parser
@hutson/semantic-delivery-gitlab > conventional-gitlab-releaser > meow > yargs-parser
@hutson/semantic-delivery-gitlab > get-pkg-repo > meow > yargs-parser
@hutson/semantic-delivery-gitlab > git-semver-tags > meow >  yargs-parser

First of all thanks for the good work on this package. The above vulnerability is currently a blocker for our CI/CD pipelines. It would be great if we can put a fix for this. Thanks!

**Comment by hutson** [comment]
Thank you for bringing this incident to my attention @rackstar.

On review it became apparent that the dependency on `meow`, and therefore `yargs-parser`, came through packages owned and maintained by the [`conventional-changelog`](https://github.com/conventional-changelog) organization.

Thankfully I still have my commit and publish access to those repositories so I was able to upgrade and publish new releases with updates to the `meow` version range.

- `get-pkg-repo` version 4.1.1 - https://github.com/conventional-changelog/get-pkg-repo/releases/tag/v4.1.1
- `conventional-commits-detector` version 1.0.3 - https://github.com/conventional-changelog/conventional-commits-detector/releases/tag/v1.0.3
- `conventional-gitlab-releaser` version 4.0.2 - https://github.com/conventional-changelog/releaser-tools/releases/tag/conventional-gitlab-releaser%404.0.2

Conducting an audit on my packages, after removing the lockfile, now returns zero known vulnerabilities:

```
semantic-delivery-gitlab$ yarn audit
yarn audit v1.22.4
info No lockfile found.
0 vulnerabilities found - Packages audited: 634
Done in 8.11s.
```

**Comment by rackstar** [comment]
Hi @hutson 

Thanks for the quick response on this issue. Vulnerabilities still come up for npm users.

```
npm audit
```

<img width="696" alt="Screenshot 2020-07-17 at 15 36 09" src="https://user-images.githubusercontent.com/15180121/87797965-51dca480-c843-11ea-8489-b88121e62de1.png">

**Comment by hutson** [comment]
Thank you @rackstar for bringing the additional vulnerability report to my attention. 

So far I have not seen this specific alert when running `npm audit` or `yarn audit` (Npm does appear to show alerts, while Yarn does not, though they use the same Audit end-point). 

Therefore, I'm slowly working my way through updating _all_ my dependencies, and I'll also take a look at `git-semver-tags` when I get a chance.

**Comment by hutson** [comment]
@rackstar after updating the project's lockfile, and several development dependencies, the audits came back clear:

```
$ npm audit

                       === npm audit security report ===

found 0 vulnerabilities
 in 654 scanned packages
```

```
$ yarn audit
yarn audit v1.13.0
0 vulnerabilities found - Packages audited: 627
Done in 1.68s.
```

I still could not reproduce the `yargs-parser` report. Could you please provide additional details on how to reproduce the same audit results?

**Comment by hutson** [comment]
I was able to finally reproduce this security alert.

```bash
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ yargs-parser                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=13.1.2 <14.0.0 || >=15.0.1 <16.0.0 || >=18.1.2             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @hutson/semantic-delivery-gitlab                             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @hutson/semantic-delivery-gitlab > git-semver-tags > meow >  │
│               │ yargs-parser                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1500                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by hutson** [comment]
Unfortunately I have not had the time to properly investigate and address the security alert reported in one of this project's deep dependencies. This reflects directly on my inability to allocate the time this project requires to make it available and useful to the community. As a result, and to avoid users from depending on an unsupported project, I am moving forward with deprecating the package on the NPM registry and archiving this repository. Please consider migrating to another supported automated release tool.
